consensus/propagation/commitment.go

```diff
@@ -119,15 +157,23 @@ func (blockProp *Reactor) handleCompactBlock(cb *proptypes.CompactBlock, peer p2
 			)
 			continue
 		}
+
+		part.Proof = *proofs[part.Index]
+
+		// note: using AddPartWithoutProof to skip verifying the proof that was
+		// generated and verified above.
 		added, err := partSet.AddPartWithoutProof(part)
 		if err != nil {
 			blockProp.Logger.Error("failed to add locally recovered part", "err", err)
 			continue
 		}
+
 		if !added {
 			blockProp.Logger.Error("failed to add locally recovered part", "part", part.Index)
 			continue
 		}
+
+		// todo: broadcast haves
```

Comment:
so we will broadcast haves for locally recovered parts? what the reasoning behind this?

Reply:
just so that we distribute the block faster. provided we have the https://github.com/celestiaorg/celestia-core/issues/1654 working, it should be secure